File: Sources/windows/yage/capabilities/sprites_provider.py
from typing import List
from yage.models import Capability, EntityState


class SpritesProvider(Capability):
    def __init__(self, subject):
        super().__init__(subject)
        subject.sprites_provider = self

    def sprite(self, state: EntityState) -> str:
        raise NotImplementedError()

    def frames(self, state: EntityState) -> List[str]:
        raise NotImplementedError()

File: Sources/macOS/Sources/Yage/Capabilities/SpritesProvider.swift
import NotAGif
import SwiftUI

open class SpritesProvider: Capability {
    open func sprite(state: EntityState) -> String {
        fatalError("Extend with your own implementation")
    }

    open func frames(state: EntityState) -> [String] {
        fatalError("Extend with your own implementation")
    }
}

public extension Entity {
    var spritesProvider: SpritesProvider? {
        capability(for: SpritesProvider.self)
    }
}
